Frends.SFTP.ListFiles/Frends.SFTP.ListFiles/Definitions/Util.cs

```diff
@@ -0,0 +1,218 @@
+    }
+
+    internal static bool IsMD5(string input)
+    {
+        if (String.IsNullOrEmpty(input))
+        {
+            return false;
+        }
+
+        return Regex.IsMatch(input, "^[0-9a-fA-F]{32}$");
+    }
+
+    internal static bool IsSha256(string input)
+    {
+        if (String.IsNullOrEmpty(input))
+        {
+            return false;
+        }
+
+        if (Regex.IsMatch(input, "^[0-9a-fA-F]{64}$"))
+            return true;
+
+        try
+        {
+            if (!input.EndsWith("="))
+                input += '=';
+            Convert.FromBase64String(input);
+            return true;
+        }
+        catch { return false; }
+    }
+
+    internal static Encoding GetEncoding(FileEncoding encoding, bool enableBom, string encodingString = null)
+    {
+        switch (encoding)
+        {
+            case FileEncoding.UTF8:
+                return enableBom ? new UTF8Encoding(true) : new UTF8Encoding(false);
+            case FileEncoding.ASCII:
+                return new ASCIIEncoding();
+            case FileEncoding.ANSI:
+                return Encoding.Default;
+            case FileEncoding.WINDOWS1252:
+                Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+                return Encoding.GetEncoding("windows-1252");
+            case FileEncoding.Other:
+                Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+                var e = Encoding.GetEncoding(encodingString);
+                if (e == null) throw new ArgumentException($"Encoding string {encodingString} is not a valid code page name.");
+                return e;
+            default:
+                throw new ArgumentOutOfRangeException($"Unknown Encoding type: '{encoding}'.");
+        }
+    }
+
+    internal static string CheckServerFingerprint(SftpClient client, string expectedServerFingerprint)
+    {
+        var userResultMessage = "";
+
+        client.HostKeyReceived += delegate (object sender, HostKeyEventArgs e)
+        {
+            if (Util.IsMD5(expectedServerFingerprint.Replace(":", "").Replace("-", "")))
+            {
+                if (!expectedServerFingerprint.Contains(':'))
+                {
+                    var serverFingerprint = BitConverter.ToString(e.FingerPrint).Replace("-", "").Replace(":", "");
+
+                    e.CanTrust = expectedServerFingerprint.ToLower() == serverFingerprint.ToLower();
+                    if (!e.CanTrust)
+                        userResultMessage = $"Can't trust SFTP server. The server fingerprint does not match. " +
+                                $"Expected fingerprint: '{expectedServerFingerprint}', but was: '{serverFingerprint}'.";
+                }
+                else
+                {
+                    var serverFingerprint = BitConverter.ToString(e.FingerPrint).Replace('-', ':');
+
+                    e.CanTrust = e.FingerPrint.SequenceEqual(Util.ConvertFingerprintToByteArray(expectedServerFingerprint));
```

Comment:
`ConvertFingerprintToByteArray` is this class method, so you can remove `Util.`-reference.